Source/sheetworkers.js

```diff
@@ -365,33 +376,127 @@ function setupDivineFavorQuery(attrs) {
+    kHarms.forEach(harm => {
+      attributes.push(`${pre}${harm}_${i}`);
+    });
   }
-  const dieFormula = dieSources
-    .map(([name, die]) => `${die}[${name}]`)
-    .join(" + ");
-  attrs[
-    "strife_formula"
-  ] = `{{roll=[[{${dieFormula}}k1 + @{strife_level}[${getTranslation(
-    "strife_level"
-  )}]]]}} {{harm=${harmType}}}`;
+  return attributes;
 }
-
 function handleStrifeRoll() {
-  getSetAttrs([...kHarms, "divine_wrath"], calcStrifeRoll, {
-    repeating: {
-      strifedie: ["strifedie_name", "strifedie_size"],
-    },
+  calcStrifeRoll();
+}
+function calcStrifeRoll() {
+  getSetAttrs(["divine_wrath"], (attrs) => {
+    attrs.repeating.characters.forEach(character => {
+      const dieSources = [];
+      const harms = [];
+      for(let i=1; i<= kSrifeDiceLength; i++) {
+        if(character[`strifedie_enabled_${i}`] && character[`strifedie_enabled_${i}`] == 1 && character[`strifedie_name_${i}`]) {
+          kHarms.forEach(harm => {
+            if(character[`${harm}_${i}`] && character[`${harm}_${i}`] == 1) {
+              harms.push(harm);
+            }
+          });
+          dieSources.push([character[`strifedie_name_${i}`], character[`strifedie_size_${i}`]]);
+        }
+      }
+      if (attrs["divine_wrath"] !== "0") dieSources.push(["@{divine_wrath_label}", "@{divine_wrath}"]);
+      const dieFormula = dieSources
+        .map(([name, die]) => `${die}[${name}]`)
+        .join(" + ");
+      const harmType = Array.from(new Set(harms)).sort();
+      const formula = `{{roll=[[{${dieFormula}}k1 + @{strife_level}[${getTranslation("strife_level")}]]]}} {{harm=${harmType}}}`;
+      character[`strife_formula`] = formula;
+    });
+  },{repeating: {characters: ["strife_formula", ...getStrifeDiceAttributes()]}});
+}
+
+function safeGuardDiceSize(roll) {
+  return roll.toLowerCase().replace(/\b[^2]?d/g, 'd');
+}
+function joinNameAndTags(characters) {
+  let joined = [];
+  characters.forEach(character => {
+    let joinedChar = {};
+    if(character.tags) {
+      joinedChar.rolls = JSON.parse(character.tags);
+      joinedChar.rolls.forEach(roll => {
+        if(roll.dice) roll.dice = safeGuardDiceSize(roll.dice);
+        if(roll.harm) roll.harm = Object.keys(roll.harm);
+      });
+    }
+    let firstTag = {};
+    firstTag.name = character.name;
+    firstTag.dice = safeGuardDiceSize(character.dice);
+    if(character.harm) firstTag.harm = Object.keys(JSON.parse(character.harm));
+    joinedChar.rolls = [firstTag, ...joinedChar.rolls];
+    if(character.notes) joinedChar.notes = character.notes;
+    joined.push(joinedChar);
+  });
+  return joined;
+}
+function handleIsland() {
+  console.log("Handling Island Drop.");
+  getSetAttrs(["drop_name", "drop_data", "drop_content", "character_id", "token_src", "token_size", "sheet_type"], (attrs, setter) => {
+    let pagedata = {};
+    try {
+      pagedata = JSON.parse(attrs.drop_data);
+    } catch(e) {
+      console.log("Invalid drop-Data attribute");
```

Comment:
```suggestion
      console.log("Invalid drop_data attribute");
```

If the data is invalid, maybe we should not try to continue from here, but instead reraise the exception with more context.